### PR TITLE
feat: add metadata.metrics.config.extraProcessors

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Signed-off-by: Dominik Rosiek <drosiek@sumologic.com>

##### Description

Fill in your description here.

---

##### Checklist

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -3764,6 +3764,18 @@ metadata:
...
+      ## It can be used for filtering metrics, renaming, changing metadata and so on.
+      ## This is list of objects, for example:
+      ## extraProcessors:
+      ##   - filterprocessor:
+      ##       exclude:
+      ##         match_type: strict
+      ##         metric_names:
+      ##           - hello_world
+      ##           - hello/world
+      extraProcessors: []
```
Comment (kkujawa-sumo) [inline deploy/helm/sumologic/values.yaml:3777]: please add information about this new parameter to https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/README.md

File: tests/helm/metrics_test.go [hunk 2]
```diff
@@ -53,3 +53,155 @@ metadata:
...
+	renameMetadatatatements := []string{
+		`set(attributes["k8s.pod.pod_name_new"], attributes["k8s.pod.pod_name"])`,
+		`delete_key(attributes, "k8s.pod.pod_name")`,
+		`set(attributes["my.static.value"], "<static_value>")`,
+	}
+	require.Equal(t, "resource", otelConfig.Processors.RenameMetadata.MetricStatements[0].Context)
+	require.Equal(t, renameMetadatatatements, otelConfig.Processors.RenameMetadata.MetricStatements[0].Statements)
+}
+
+func TestMetadataMetricsOtelConfigExtraProcessorsPipeline(t *testing.T) {
```
Comment (swiatekm-sumo) [inline tests/helm/metrics_test.go:141]: These two tests use the same configuration, could we merge them into one? The actual assertions in this test aren't very complex, adding them to the first test should be ok.

I'd also consider moving the valuesYaml to a file in `testdata/`, it's a bit large for an inline value imo.

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/values.yaml:3777]: I wonder if this shouldn't live under `sumologic.metrics.metadata` instead. :thinking: It involves OT config internals, but I imagine it's also going to be one of the main ways of customizing the configuration for users.

File: docs/collecting-application-metrics.md [hunk 3]
```diff
@@ -209,3 +220,127 @@ To scrape and forward exposed metrics to Sumo Logic, please follow one of the fo
...
+
+This section coverts the following metrics modifications:
+
+- [Filtering metrics](#filtering-metrics)
+- [Renaming metric](#renaming-metric)
+- [Adding or renaming metadata](#adding-or-renaming-metadata)
+
+### Filtering metrics
+
+In order to filter in or out the metrics, you can add [filterprocessor] to metric's extraProcessors.
```
Comment (swiatekm-sumo) [inline docs/collecting-application-metrics.md:234]: Missing the link for `[filteprocessor]`.

File: docs/collecting-application-metrics.md [hunk 4]
```diff
@@ -209,3 +220,127 @@ To scrape and forward exposed metrics to Sumo Logic, please follow one of the fo
 
 - [Application metrics are exposed (one endpoint scenario)](#application-metrics-are-exposed-one-endpoint-scenario)
 - [Application metrics are exposed (multiple enpoints scenario)](#application-metrics-are-exposed-multiple-enpoints-scenario)
+
+## Metrics modifications
```
Comment (swiatekm-sumo) [inline docs/collecting-application-metrics.md:224]: Can you also update https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/best-practices.md#excluding-metrics and other metrics sections in the best practices doc with this new info?

Comment (sumo-drosiek) [reply docs/collecting-application-metrics.md:224]: Should I write more information there or linking to this document will be sufficient?

Comment (swiatekm-sumo) [reply docs/collecting-application-metrics.md:224]: I'd provide some examples. It's structurally not the same thing either, as you may want to filter and modify non-application metrics.

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:3777]: I was considering `metadata.metrics` as Otel specific configuration and extraProcessors is definitely Otel specific

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml:3808]: I guess it's ok until we expose simpler higher-level ways of doing this, like namespace regexes and so forth.

File: tests/helm/metrics_test.go (hunk 2 above)
Comment (sumo-drosiek) [reply tests/helm/metrics_test.go]: done

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**